src/core/tensor/math_kernel.cu

```diff
@@ -140,7 +140,12 @@ __global__ void KernelRound(const size_t n, const float *in, float *out) {
 __global__ void KernelRoundE(const size_t n, const float *in, float *out) {
   for (int i = blockIdx.x * blockDim.x + threadIdx.x; i < n;
        i += blockDim.x * gridDim.x) {
-    out[i] = roundf(in[i]/2)*2;
+    auto doub = in[i]*2;
```

Comment:
This auto causes problem in C complier version 5.4, upgraded to 7.5.0 will then be okay. Resolved in #759

Reply:
Thank you Chris for the prompt response! I could also replace the `auto` in this PR. Yes I think updating to newer gcc is a good idea, although the "new"(c++11) one is already widely used and not that new, it would be helpful to support running c++11 code.

Reply:
I guess there is no need to change the `auto`, `auto` seems to be a good way to prevent implicit type casting (sometime developer us could be careless). The only problem of `auto` is just the code readability for the developer to know the exact datatype.

In any case, I think need to keep the same version of complier between the docker version and conda version (as in #759), because a few day ago we had the similar complier error because of the version different.